components/legacy_team_sidebar/components/team_button.tsx

```diff
@@ -1,77 +1,65 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {Tooltip} from 'react-bootstrap';
-import {injectIntl} from 'react-intl';
+import {injectIntl, IntlShape} from 'react-intl';
 import {Link} from 'react-router-dom';
 import {Draggable} from 'react-beautiful-dnd';
 import classNames from 'classnames';
 
 import {mark, trackEvent} from 'actions/diagnostics_actions.jsx';
 import Constants from 'utils/constants';
-import {intlShape} from 'utils/react_intl';
 import {isDesktopApp} from 'utils/user_agent';
-import {localizeMessage, isMac} from 'utils/utils.jsx';
+import {isMac, localizeMessage} from 'utils/utils.jsx';
 import CopyUrlContextMenu from 'components/copy_url_context_menu';
 import OverlayTrigger from 'components/overlay_trigger';
 import TeamIcon from '../../widgets/team_icon/team_icon';
 
+interface Props extends Pick<any, any>{
```

Comment:
Why the need for Pick here?

Reply:
Without that, I'm getting the following error.  
```
error TS2740: Type '{ btnClass: string; key: string; url: string; tip: Element; content: string; switchTeam: Function; }' is missing the following properties from type 'Pick<Props, "content" | "disabled" | "displayName" | "url" | "placement" | "btnClass" | "active" | "teamId" | "tip" | "order" | "showOrder" | "unread" | "mentions" | "teamIconUrl" | "switchTeam" | "isDraggable" | "teamIndex">': disabled, displayName, placement, active, and 8 more.

```
I don't know how to solve this. If you could help me in solving it then